docs/install/authentication-mechanisms-gcp.md

```diff
@@ -131,3 +131,121 @@ gcloud projects add-iam-policy-binding $PROJECT_ID \
+    ```
+    if the annotation exists, you'll find an annotation similar to:
+    ```shell
+    metadata:
+      annotations:
+        iam.gke.io/gcp-service-account: gsa-name@$PROJECT_ID.iam.gserviceaccount.com
+    ```
+
+### Kubernetes Secrets
+
+To ensure kubernetes Secret correctly configured in a namespace, you need to check if a Kubernetes Secret 
+named `google-cloud-key` is in desired namespace by running the following command: 
+
+```shell
+kubectl get secret -n secret-namespace
+```
+
+### Common Issues
+
+* ***Resources are not READY***
+
+   If a resource instance is not READY due to authentication configuration problem, 
+   you are likely to get `PermissionDenied` error for `User not authorized to perform this action`
+   (for Workload Identity, the error might look like `IAM returned 403 Forbidden: The caller does not have permission`).
+   
+   This error indicates that the Control Plane may not configure authentication properly. You can find it in a resource instance by:
+   
+   ```shell
+   kubectl describe resource-name resource-instance-name -n resource-instance-namespace
+   ```
+   
+   Here is an example checking a CloudAuditlogsSource `test` in namespace `default`:
+      
+   ```shell
+   kubectl describe cloudauditlogssource test -n default
+   ```
+   
+   To solve this issue, you can:
+   - Check the Google Cloud Service Account `cloud-run-events` for the Control Plane has all required permissions.
+   - Check authentication configuration is correct for the Control Plane. 
+     - If you are using Workload Identity for the Control Plane, 
+     refer [here](./authentication-mechanisms-gcp.md/#workload-identity) to check the
+     Google Cloud Service Account `cloud-run-events`, 
+     and the Kubernetes Service Account `controller` in namespace `cloud-run-events`.
+     - If you are using Kubernetes Secret for the Control Plane, 
+     refer [here](./authentication-mechanisms-gcp.md/#kubernetes-secrets) to check the 
+     Kubernetes Secret `google-cloud-key` in namespace `cloud-run-events`.
+     
+   ***Note:*** For Kubernetes Secret, it is also possible that the JSON private key is no longer existing under your Google Cloud Service Account 
+   `cloud-run-events`. Then, even the Google Cloud Service Account `cloud-run-events` has all required permissions, and 
+   the corresponding Kubernetes Secret `google-cloud-key ` is in namespace `cloud-run-events`, you still get `PermissionDenied` error. 
+   To such case, you have to re-download the JSON private key and re-create the Kubernetes Secret.
+     
+* ***Resources are READY, but can't receive Event***
+
+   Sometimes, a resource instance is READY, but it can't receive Event. It might be authentication configuration problem, 
+   if the underlying `deployment` doesn't have minimum availability.
+   
+   You can use the following command to check a `deployment`'s available pod is zero.
+   
+   ```shell
+   kubectl get deployment -n resource-instance-namespace
```

Comment:
Can we tell them which Deployment to look at?

Reply:
We can (at some levels), the deployment name is a truncated combination of prefix, Namespace, Name, and UID. I'll update it.